assets/javascripts/kitten/karl/images/cropping-image.js

```diff
@@ -90,20 +99,31 @@ export class KarlCroppingImage extends React.Component {
 
   handleCrop() {
     if (this.state.imageSrc) {
-      const dataUrl = this.refs.cropper.getCroppedCanvas().toDataURL()
+      const croppedCanvas = this.refs.cropper.getCroppedCanvas()
 
-      this.setState({
-        imageCropSrc: dataUrl,
-      })
+      if (croppedCanvas) {
+        this.setState({
+          imageCropSrc: croppedCanvas.toDataURL(),
+        })
+      }
     }
   }
 
+  setCropperHeight() {
+    const width = domElementHelper.getComputedWidth(this.refs.cropperContainer)
+    const height = width * 9 / 16
+
+    this.setState({
+      cropperHeight: height,
+    })
+  }
+
   renderCropper() {
     const cropperProps = {
       ref: 'cropper',
```

Comment:
Il y a une nouvelle syntaxe pour faire des références à des éléments. `<Cropper ref={ e => this._cropper = e } } …/>` et on accède à la prop en faisant `this._cropper` plutôt que `this.refs.cropper`.